src/Glpi/Config/LegacyConfigurators/BaseConfig.php

```diff
@@ -0,0 +1,274 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Config\LegacyConfigurators;
+
+use Glpi\Config\LegacyConfigProviderInterface;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\HttpFoundation\Request;
+
+final readonly class BaseConfig implements LegacyConfigProviderInterface
+{
+    public function __construct(
+        #[Autowire('%kernel.project_dir%')] private string $projectDir,
+    ) {
+    }
+
+    public function execute(Request $request): void
+    {
+        if (!defined('GLPI_ROOT')) {
+            define('GLPI_ROOT', $this->projectDir);
+        }
+
+        // Notice problem  for date function :
+        $tz = ini_get('date.timezone');
+        if (!empty($tz)) {
+            date_default_timezone_set($tz);
+        } else {
+            date_default_timezone_set(@date_default_timezone_get());
+        }
+
+        include_once(GLPI_ROOT . "/inc/autoload.function.php");
+
+        // Check if web root is configured correctly
+        if (!isCommandLine()) {
+            $included_files = array_filter(
+                get_included_files(),
+                function (string $included_file) {
+                    // prevent `tests/router.php` to be considered as initial script
+                    return realpath($included_file) !== realpath(sprintf('%s/tests/router.php', GLPI_ROOT));
+                }
+            );
+
+            $initial_script = array_shift($included_files) ?? __FILE__;
+
+            // If `auto_prepend_file` configuration is used, ignore first included files
+            // as long as they are not located inside GLPI directory tree.
+            $prepended_file = ini_get('auto_prepend_file');
+            if ($prepended_file !== '' && $prepended_file !== 'none') {
+                $prepended_file = stream_resolve_include_path($prepended_file);
+                while (
+                    $initial_script !== null
+                    && !str_starts_with(
+                        realpath($initial_script) ?: '',
+                        realpath(GLPI_ROOT)
+                    )
+                ) {
+                    $initial_script = array_shift($included_files);
+                }
+            }
+
+            if (realpath($initial_script) !== realpath(sprintf('%s/public/index.php', GLPI_ROOT))) {
+                echo sprintf(
+                    'Web server root directory configuration is incorrect, it should be "%s". See installation documentation for more details.' . PHP_EOL,
+                    realpath(sprintf('%s/public', GLPI_ROOT))
+                );
+                exit(1);
+            }
+        }
```

Comment:
This can be removed now. As long as all the legacy bootstraping logic is removed from `inc/xxx.php` files, it is not anymore possible to execute GLPI without using the `public/index.php` file.
```suggestion
```